client/boot/app.js

```diff
@@ -42,7 +42,7 @@ const boot = currentUser => {
 		setupMiddlewares( currentUser, reduxStore );
 		invoke( project, 'setupMiddlewares', currentUser, reduxStore );
 		detectHistoryNavigation.start();
-		page.start();
+		page.start( { decodeURLComponents: false } );
```

Comment:
I did a quick scan for anything using `querystring` in calypso and couldn't find a case where this was an issue